File: languages/r36/template/src/Algorithm.r
library(algorithmia)
# API calls will begin at the algorithm() method, with the request body passed as 'input'
# For more details, see algorithmia.com/developers/algorithm-development/languages

algorithm <- function(input) {
  paste("hello", input)
}

# Here is an example of an advanced form of an algorithm function, using a load function (like the example below)
# -- ADVANCED ALGORITHM USAGE --
#   algorithm <- function(input, keras_model) {
#   ....
#   prediction <- keras_model.predict(input)
#   result <- list(class <- prediction[0], confidence <- prediction[1])
#   result
# }


load <- function() {
  # Here you can optionally define a function called when the algorithm is loaded.
  # A great example would be any model files that need to be available to this algorithm
  # during runtime.
  # Any variables returned here, will be passed as the secondary argument to your 'algorithm' function
  
  # -- USAGE EXAMPLE ---
  # client <- getAlgorithmiaClient()
  # model_file_path <- client$file('data://path/to/my/modelFile.hd5)$getFile()$name
  # keras_model <- keras.load_model(model_path)
  # keras_model
  NULL
}

# This code turns your library code into an algorithm that can run on the platform.
# If you intend to use loading operations, remember to pass a `load` function as a second variable.
algo <- getAlgorithmHandler(algorithm, load)
# The 'serve()' function actually starts the algorithm, you can follow along in the source code.
algo$serve()

File: languages/python3-adk/template/src/Algorithm.py
from Algorithmia import ADK


# API calls will begin at the apply() method, with the request body passed as 'input'
# For more details, see algorithmia.com/developers/algorithm-development/languages

def apply(input, globals):
    # If your apply function uses state that's loaded into memory via load, you can pass that loaded state to your apply
    # function by defining an additional "globals" parameter in your apply function.
    return "hello {} {}".format(str(input), str(globals['payload']))


def load():
    # Here you can optionally define a function that will be called when the algorithm is loaded.
    # The return object from this function can be passed directly as input to your apply function.
    # A great example would be any model files that need to be available to this algorithm
    # during runtime.
    # Any variables returned here, will be passed as the secondary argument to your 'algorithm' function
    globals = {}
    globals['payload'] = "Loading has been completed."
    return globals


# This turns your library code into an algorithm that can run on the platform.
# If you intend to use loading operations, remember to pass a `load` function as a second variable.
algorithm = ADK(apply, load)
# The 'init()' function actually starts the algorithm, you can follow along in the source code
# to see how everything works.
algorithm.init("Algorithmia")
